cadc-conesearch/src/main/java/org/opencadc/conesearch/TAPQueryGenerator.java

```diff
@@ -0,0 +1,236 @@
+ *
+ *
+ ************************************************************************
+ */
+
+package org.opencadc.conesearch;
+
+import ca.nrc.cadc.dali.Circle;
+import ca.nrc.cadc.dali.CommonParamValidator;
+import ca.nrc.cadc.dali.tables.votable.VOTableWriter;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+
+/**
+ * Class to produce an ADQL query from a set of Parameters
+ */
+public class TAPQueryGenerator extends CommonParamValidator {
+    private static final int DEF_MAXREC = 1000;
+    private static final int MAX_MAXREC = Integer.MAX_VALUE;
+    private static final int MIN_VERB_VALUE = 1;
+    private static final int MID_VERB_VALUE = 2;
+    private static final int MAX_VERB_VALUE = 3;
+    private static final Logger LOGGER = LogManager.getLogger(TAPQueryGenerator.class);
+
+    protected final String catalog;
+    protected final String lowVerbositySelectList;
+    protected final String midVerbositySelectList;
+    protected final String highVerbositySelectList;
+
+    public TAPQueryGenerator(final String catalog, final String lowVerbositySelectList,
+                             final String midVerbositySelectList, final String highVerbositySelectList) {
+        this.catalog = catalog;
+        this.lowVerbositySelectList = lowVerbositySelectList;
+        this.midVerbositySelectList = midVerbositySelectList;
+        this.highVerbositySelectList = highVerbositySelectList;
+    }
+
+    /**
+     * Map with the REQUEST, LANG, and QUERY parameters.
+     *
+     * @param positionColumnName The name of the positional column to compare against the cone's position,
+     *                           preferably spatially indexed.
+     * @return map of parameter names and values
+     */
+    public Map<String, Object> getParameterMap(final String positionColumnName,
+                                               final Map<String, List<String>> parameters) {
+        final Map<String, Object> queryParameterMap = new HashMap<>();
+
+        // Obtain and, if necessary, provide a default RESPONSEFORMAT.
+        final String requestedResponseFormat = getFirstParameter(ParameterNames.RESPONSEFORMAT.name(), parameters,
+                                                                 VOTableWriter.CONTENT_TYPE);
+        queryParameterMap.put(ParameterNames.RESPONSEFORMAT.name(), requestedResponseFormat);
+
+        // Obtain and validate the VERB (verbosity) output.
+        final String requestedOutputVerbosity = getFirstParameter(ParameterNames.VERB.name(), parameters,
+                                                                  Integer.toString(MID_VERB_VALUE));
+        final NumberParameterValidator outputVerbosityValidator =
+                new NumberParameterValidator(false, TAPQueryGenerator.MIN_VERB_VALUE,
```

Comment:
rather than a separate NumberParameterValidator, just use `CommonParamValidator.validateInteger(...)`. You'll have to deal with getting the first value from the list here (multi-valued params are the norm now so not generically supported in CommonParamValidator)

Reply:
This is done.